File: challenges/30-days-of-code/day1-DataTypes/datatype.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {
	
    public static void main(String[] args) {
        int i = 4;
        double d = 4.0;
        String s = "HackerRank ";
		
        Scanner scan = new Scanner(System.in);

        /* Declare second integer, double, and String variables. */
        int integer;
        double doub;
        String str;

        /* Read and save an integer, double, and String to your variables.*/
        // Note: If you have trouble reading the entire String, please go back and                  review the Tutorial closely.
        integer = scan.nextInt(); 
        doub = scan.nextDouble();
        scan.nextLine(); //finishes scanning the second line 
        str = scan.nextLine();

        /* Print the sum of both integer variables on a new line. */
        /* Print the sum of the double variables on a new line. */
		System.out.println(i + integer);
        System.out.println(d + doub);

        /* Concatenate and print the String variables on a new line; 
        	the 's' variable above should be printed first. */
        System.out.println(s + str);

    
        scan.close();
    }
}

File: challenges/30-days-of-code/day1-DataTypes/datatype.cpp
#include <iostream>
#include <iomanip>
#include <limits>

using namespace std;

int main() {
    int i = 4;
    double d = 4.0;
    string s = "HackerRank ";

    
    // Declare second integer, double, and String variables.
    int integer = 0;
    double dub = 0.0;
    string str = " ";

    // Read and save an integer, double, and String to your variables.
    cin >> integer;
    cin >> dub;
    cin.ignore();  //ignores an end of line character after cin leaves a char at the end
    getline(cin,str);

    // Print the sum of both integer variables on a new line.
    cout << integer + i << endl;
    // Print the sum of the double variables on a new line.
    cout.precision(1);
    cout << fixed << d + dub << endl;
    // Concatenate and print the String variables on a new line
    // The 's' variable above should be printed first.
    cout << s << "" << str << endl; 


    return 0;
}